### DISCUSSION TITLE
Initializing model with position dependent values

### DISCUSSION THREAD

**Discussion by philomenaweng** [pybamm-team/PyBaMM#3833] [Q&A] [answered]
Is there a way to initialize a DFN model with a position-dependent values (i.e. x-dependent porosity or SEI thickness)?

**Comment by valentinsulzer** [comment, marked as answer]
Porosity, active material fraction, initial concentration: yes. Just pass in a function
```python
def eps(x):
    return ...

parameter_values["Separator porosity"] = eps
```
The function takes in `x` in meters relative to the anode current collector.

SEI thickness: not right now but it could be added fairly easily.

**Reply by philomenaweng** [reply]
thank you!!
